src/controller.rs

```diff
@@ -498,6 +542,31 @@ fn pod_template(config: &WasmCloudHostConfig, _ctx: Arc<Context>) -> PodTemplate
     template
 }
 
+async fn discover_configmap_certificates(
+    namespace: String,
+    name: String,
+    ctx: &Context,
+) -> Result<Vec<String>> {
+    let kube_client = ctx.client.clone();
+    let api = Api::<ConfigMap>::namespaced(kube_client, &namespace);
+    let mut certs = Vec::new();
+
+    let raw_config_map = api.get(name.as_str()).await.map_err(|e| {
+        warn!("Failed to get configmap for certificate discovery: {}", e);
+        e
+    })?;
```

Comment:
Assuming the changes to the arguments to this function (i.e. changing from `String` to `&str`), you can adapt this code to read as follows:
```suggestion
    let kube_client = ctx.client.clone();
    let api = Api::<ConfigMap>::namespaced(kube_client, namespace);
    let mut certs = Vec::new();

    let raw_config_map = api.get(name).await.map_err(|e| {
        warn!("Failed to get configmap for certificate discovery: {}", e);
        e
    })?;
```
Which I think has the added benefit of reading more nicely 🙂

Reply:
It's also worth noting, that when you are casting a `String` to `&str`, you can simply use `&<string_variable_name>`, no need for `.as_str()` 🙂